source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -293,7 +302,53 @@ void SslSocket::onConnected() { ASSERT(info_->state() == Ssl::SocketState::PreHa
 
 Ssl::ConnectionInfoConstSharedPtr SslSocket::ssl() const { return info_; }
 
+void SslSocket::keylogCallback(const SSL* ssl, const char* line) {
+  if (ssl == nullptr) {
+    return;
+  }
+  BIO* bio_keylog = static_cast<BIO*> SSL_get_app_data(ssl);
+  BIO_printf(bio_keylog, "%s\n", line);
+  (void)BIO_flush(bio_keylog);
+}
+
+int SslSocket::setSSLKeyLog(bool enable) {
```

Comment:
I think we don't have to return any value because it seems we don't use it. such as ssl_socket.cc#L70